Entity/BaseUser.php

```diff
@@ -232,4 +235,26 @@ public function canSwitchTo(BaseUser $user): bool
     {
         return false;
     }
+
+    public function serialize(): string
+    {
+        return serialize(
+            [
+                $this->id,
+                $this->username,
+                $this->password,
```

Comment:
Are you 100% sure this should be serialised?

Reply:
Can you not just serialize id, username and email? Password shouldn't be serialized (definitely not with salt).

Reply:
According to the documentation I read, password and salt can also be serialized. Actually, if we don't override the serialization of the user like I do in this PR, all fields get serialized (password and salt included) - that's the Symfony way.
Nevertheless I tested it without those fields and it works, so I guess they aren't necessary.